src/main/java/nwoolcan/model/brewery/warehouse/Warehouse.java

```diff
@@ -33,22 +36,13 @@
      */
     List<Article> getArticles(QueryArticle queryArticle);
     /**
-     * Add a record to the warehouse: creates a {@link Stock} if it doesn't exist.
-     * It may fails if the {@link Record} is not compatible with the {@link Article}.
-     * @param article which is moving in or out the warehouse.
-     * @param expirationDate of the article, if present, to identify the right {@link Stock}.
-     * @param record to be registered
-     * @return a {@link Result} of {@link Empty} which reports possible errors.
-     */
-    Result<Empty> addRecord(Article article, Date expirationDate, Record record);
-    /**
-     * Add a record to the warehouse: creates a {@link Stock} if it doesn't exist.
-     * It may fails if the {@link Record} is not compatible with the {@link Article}.
-     * @param article which is moving in or out the warehouse.
+     * Add a record to the {@link Stock} in the {@link Warehouse}.
+     * It may fails if the {@link Record} is not compatible with the {@link Article} in the {@link Stock}.
+     * @param stock on which effect the transaction.
      * @param record to be registered.
      * @return a {@link Result} of {@link Empty} which reports possible errors.
      */
-    Result<Empty> addRecord(Article article, Record record);
+    Result<Empty> addRecord(Stock stock, Record record);
```

Comment:
Why is this method here? Shouldn't it be a `Stock` method?
I mean, there is the same method in `Stock`, why need to do further checks since you cannot create a `Stock` from out of the `Warehouse`?

Reply:
good point.. i was still at previous approach were I could not assume `Stock` were created only from `Warehouse`